modules/renter/renter.go

```diff
@@ -177,6 +180,16 @@ func newRenter(cs modules.ConsensusSet, tpool modules.TransactionPool, hdb hostD
 	go r.threadedRepairLoop()
 	go r.threadedDownloadLoop()
 	go r.threadedQueueRepairs()
+
+	// Kill workers on shutdown.
+	r.tg.OnStop(func() {
+		id := r.mu.RLock()
+		for _, worker := range r.workerPool {
+			close(worker.killChan)
+		}
+		r.mu.RUnlock(id)
+	})
```

Comment:
I think there is actually a deadlock.
The worker loop looks like:
```go
for {
		select {
		case <-w.killChan:
			return
		default:
		}
		if w.renter.tg.Add() != nil {
			return
		}
		w.work()
		w.renter.tg.Done()
}
```
But `tg.Add` and `tg.Stop` both lock `tg.bmu`, so `tg.Add` will block until shutdown finishes.